src/hotspot/os/aix/os_aix.cpp

```diff
@@ -295,6 +296,8 @@ static bool my_disclaim64(char* addr, size_t size) {
 
   for (unsigned int i = 0; i < numFullDisclaimsNeeded; i ++) {
     if (::disclaim(p, maxDisclaimSize, DISCLAIM_ZEROMEM) != 0) {
+      ErrnoPreserver ep;
+      log_trace(os,map)("Cannot disclaim %p - %p (errno %d)\n", p, p + maxDisclaimSize, errno);
```

Comment:
Using `os::strerror` or `os::errno_name` would be more user friendly than printing the numeric value. Don't simply replicate the crude approach of `trcVerbose`. This applies everywhere `errno` is being printed.

Reply:
Also isn't the whole point of using the preserver to allow you capture errno and read it back from the preserver without caring if something in the logging code changed it in the meantime?

Reply:
i.e rather than passing `errno` as a parameter, you should be passing `ep.errno()`

Reply:
> crude approach of trcVerbose

Opinions differ, obviously, since these trace lines are from me originally. Not sure what's crude about them.

I don't care much one way or the other, but just to defend my original code: 
- errno numerical are short, concise and well defined within their OS. 
- their enum counterparts (EINVAL etc) are too
- strerror is not as clear and, most annoyingly, localization-dependent. A german customer running the JVM with a german locale will see (and give to your support people) logs containing german error messages.

Reply:
> i.e rather than passing `errno` as a parameter, you should be passing `ep.errno()`

That is right, I did not think of that. By the time the raw errno is passed to the logging function, it may have been changed already. That is very improbable, but could happen.

E.g. 
```
log_info(os)("Something happen (%s) (errno=%d)", some_function_that_changes_errno_and_returns_string(), errno);
```